app/models/goal.py

```diff
@@ -1,5 +1,12 @@
 from app import db
-
-
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String, nullable=False)
+    tasks = db.relationship('Task', backref='goal', lazy=True)
+
+    def goal_response_body_dict(self):
```

Comment:
Style: Since this is a method on `Goal` you don't need `goal` in the name:

```suggestion
    def goal_response_body_dict(self):
```
```suggestion
    def response_body_dict(self):
```